contracts/Everlong.sol

```diff
@@ -66,64 +67,313 @@ import { EVERLONG_KIND, EVERLONG_VERSION } from "./libraries/Constants.sol";
+    /// @param _targetOutput Minimum amount of proceeds to receive.
+    /// @return output Total output received from closed positions.
+    function _closePositions(
+        uint256 _targetOutput
+    ) internal returns (uint256 output) {
+        while (!_portfolio.isEmpty() && output < _targetOutput) {
+            output += _hyperdrive.closeLong(_asBase, _portfolio.head());
+            _portfolio.handleClosePosition();
+        }
+        return output;
     }
 
     // ╭─────────────────────────────────────────────────────────╮
     // │ Getters                                                 │
     // ╰─────────────────────────────────────────────────────────╯
 
-    /// @notice Gets the Everlong instance's kind.
-    /// @return The Everlong instance's kind.
-    function kind() external view returns (string memory) {
-        return EVERLONG_KIND;
+    /// @notice Name of the Everlong token.
+    /// @return Name of the Everlong token.
+    function name() public view override returns (string memory) {
+        return _name;
+    }
+
+    /// @notice Symbol of the Everlong token.
+    /// @return Symbol of the Everlong token.
+    function symbol() public view override returns (string memory) {
+        return _symbol;
     }
 
-    /// @notice Gets the Everlong instance's version.
-    /// @return The Everlong instance's version.
-    function version() external view returns (string memory) {
-        return EVERLONG_VERSION;
+    /// @dev The underlying asset decimals.
+    /// @return The underlying asset decimals.
+    function _underlyingDecimals()
+        internal
+        view
+        virtual
+        override
+        returns (uint8)
+    {
+        return _decimals;
     }
 
-    /// @notice Gets the address of the underlying Hyperdrive Instance.
-    function hyperdrive() external view returns (address) {
-        return _hyperdrive;
+    /// @notice The address of the underlying Hyperdrive Instance.
+    /// @return The address of the underlying Hyperdrive Instance.
+    function hyperdrive() external view override returns (address) {
+        return address(_hyperdrive);
     }
 
-    /// @notice Gets whether Everlong uses Hyperdrive's base token to
-    ///         transact.
+    /// @notice Whether Everlong uses Hyperdrive's base token to transact.
+    /// @return Whether Everlong uses Hyperdrive's base token to transact.
     function asBase() external view returns (bool) {
         return _asBase;
     }
+
+    /// @notice Address of the token used to interact with the Hyperdrive instance.
+    /// @return Address of the token used to interact with the Hyperdrive instance.
+    function asset() public view override returns (address) {
+        return address(_asset);
+    }
+
+    /// @notice Gets the admin address for Everlong.
+    function admin() external view returns (address) {
+        return _admin;
+    }
+
+    // FIXME: Consider idle liquidity + maybe maxLong?
```

Comment:
Is the plan to fix this in this PR or a follow-up?

Reply:
Yep I commented a list of todos on the "Everlong" issue, not final by any means but should overview what is to come